### PR TITLE
lint/vet/clean

### PR DESCRIPTION
**Pull request by hydroflame**

### CODE REVIEW

File: vcs.go [hunk 1]
```diff
@@ -110,59 +115,12 @@ func (g Git) OpenFile(revision, path string) (io.ReadCloser, error) {
...
+func (g *Git) DefaultRevision() (string, string) {
 	// Check if there's unstaged changes, if so, return dot
 	contents, _ := exec.Command("git", "--git-dir", g.dir, "ls-files", "-m").Output()
 	if len(contents) > 0 {
 		return "HEAD", "."
 	}
 	return "HEAD~1", "HEAD"
 }
-
-type fileInfo struct {
```
Comment (bradleyfalzon) [inline vcs.go]: fileInfo was still needed in git `ReadDir`, I don't think it can be moved to to the test file.

Comment (hydroflame) [inline vcs.go]: odd, go test github.com/hydroflame/abicheck && go install github.com/hydroflame/abicheck/cmd/abicheck both pass

Comment (hydroflame) [inline vcs.go]: but you're right, wtf is happening on machine

File: vcs.go [hunk 2]
```diff
@@ -110,59 +115,12 @@ func (g Git) OpenFile(revision, path string) (io.ReadCloser, error) {
...
-}
-
-func (fi fileInfo) Name() string       { return fi.name }
-func (fi fileInfo) Size() int64        { panic("not implemented") }
-func (fi fileInfo) Mode() os.FileMode  { panic("not implemented") }
-func (fi fileInfo) ModTime() time.Time { panic("not implemented") }
-func (fi fileInfo) IsDir() bool        { return fi.dir }
-func (fi fileInfo) Sys() interface{}   { panic("not implemented") }
-
-var _ VCS = (*StrVCS)(nil)
```
Comment (bradleyfalzon) [inline vcs.go]: I use StrVCS in abicheck-web (https://github.com/bradleyfalzon/abicheck-web/blob/master/main.go#L46) as I didn't want to combine the two projects.

File: vcs.go [hunk 3]
```diff
@@ -92,7 +96,8 @@ func (g Git) ReadDir(revision, path string) ([]os.FileInfo, error) {
 	return files, nil
 }
 
-func (g Git) OpenFile(revision, path string) (io.ReadCloser, error) {
+// OpenFile returns a reader for a given absolute path at a revision
+func (g *Git) OpenFile(revision, path string) (io.ReadCloser, error) {
```
Comment (bradleyfalzon) [inline vcs.go]: I'm curious why this should be changed to a pointer receiver? I didn't think it needed to be?

Comment (hydroflame) [inline vcs.go]: Doesn't *need* to be, but you're returning a *Git so all your functions should be pointer receivers for consistency

**Comment by bradleyfalzon** [comment]
you have some messages from me on slack btw, not related to this pr

File: abicheck_test.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ import (
 // TestParse tests the results from the parser against an expected golden master
 func TestParse(t *testing.T) {
 	// Create strvcs and fill it with test data
-	vcs := StrVCS{}
+	var vcs strVCS
```
Comment (bradleyfalzon) [inline abicheck_test.go]: This also needs to refer to the reexported name.

**Comment by bradleyfalzon** [comment]
Thanks for the review, can you squash to the one commit and then I'll merge

**Comment by bradleyfalzon** [comment]
Merged via 4a98885129f33e59ca24b288ffb199b5cb78ed69

**Pull request closed without merging**